### PR TITLE
Add rstcheck linter to check for errors in restructured text

### PR DESCRIPTION
**Pull request by jnduli**

### CODE REVIEW

File: ale_linters/rst/rstcheck.vim [hunk 1]
```diff
@@ -0,0 +1,30 @@
+" Author: John Nduli https://github.com/jnduli
+" Description: Rstcheck for reStructuredText files
+
+let g:ale_lint_delay = 1000
```
Comment (w0rp) [inline ale_linters/rst/rstcheck.vim]: You left this line lying around here.

File: ale_linters/rst/rstcheck.vim [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    " mismatch.'
+    let l:pattern = '\v^\S*:(\d*): \(([a-zA-Z]*)/\d*\) (.+)$'
+    let l:output = []
+    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        call add(l:output, {
+                    \   'lnum': l:match[1] + 0,
+                    \   'col': 0,
+                    \   'type': l:match[2] is# 'SEVERE' ? 'E' : 'W',
+                    \   'text': l:match[3],
+                    \})
```
Comment (w0rp) [inline ale_linters/rst/rstcheck.vim]: Move the line continuation to the same indentation level as previous lines. Have a look through existing code in the codebase.

File: test/handler/test_rstcheck_lint_handler.vader [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  \      'col': 0,
+  \      'type': 'E',
+  \      'text': 'Title overline & underline mismatch.',
+  \   },
+  \ ],
+  \ ale_linters#rst#rstcheck#Handle(1, [
+  \   'foo.rst',
+  \   'bad_python.rst:7: (ERROR/3) (python) unexpected EOF while parsing',
+  \   'bad_cpp.rst:9: (ERROR/3) (cpp) error: ''x'' was not declared in this scope',
+  \   'bad_rst.rst:1: (SEVERE/4) Title overline & underline mismatch.',
```
Comment (w0rp) [inline test/handler/test_rstcheck_lint_handler.vader]: Will the linter return the filenames from many files, such as errors from imported files? If so, you should add the filenames to a match group, figure out what the absolute paths are, and include them in the items with a `filename` key.

File: ale_linters/rst/rstcheck.vim (hunk 1 above)
Comment (jnduli) [reply ale_linters/rst/rstcheck.vim]: I'll remove this. Thanks.

File: ale_linters/rst/rstcheck.vim (hunk 2 above)
Comment (jnduli) [reply ale_linters/rst/rstcheck.vim]: I'll look through them and update this.

File: test/handler/test_rstcheck_lint_handler.vader (hunk 3 above)
Comment (jnduli) [reply test/handler/test_rstcheck_lint_handler.vader]: When you say filename, do you mean I include it in the returns from the handler. similar to lnum and col?

File: ale_linters/rst/rstcheck.vim [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+" Description: Rstcheck for reStructuredText files
+
+function! ale_linters#rst#rstcheck#Handle(buffer, lines) abort
+    " matches: 'bad_rst.rst:1: (SEVERE/4) Title overline & underline
+    " mismatch.'
+    let l:pattern = '\v^(.+):(\d*): \(([a-zA-Z]*)/\d*\) (.+)$'
+    let l:output = []
+    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        call add(l:output, {
+        \   'filename': l:match[1],
```
Comment (w0rp) [inline ale_linters/rst/rstcheck.vim]: Does `rstcheck` return relative paths or absolute paths?

If the paths are relative, you'll have to figure out what they are relative to, and then compute absolute paths from that with `ale#path#GetAbsPath`. If they are just relative to the working directory, prefix the command with `cd directory_for_file &&`. There are a few other linters that do this with the `ale#path#BufferCdString` function.

**Pull request merged**